.github/workflows/crystal.yml

```diff
@@ -2,31 +2,57 @@ name: CI
 
 on:
   push:
-  workflow_dispatch:
   schedule:
     - cron: "0 6 * * 1"
 
 jobs:
-  build:
+  style:
+    runs-on: ubuntu-latest
+    container: crystallang/crystal
+    steps:
+      - uses: actions/checkout@v2
+      - name: Format
+        run: crystal tool format --check
+      - name: Lint
+        uses: crystal-ameba/github-action@v0.2.12
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  test:
     strategy:
       fail-fast: false
       matrix:
-        experimental: [false]
-        os: [ubuntu-latest]
+        stable: [true]
         crystal:
-          - latest
+          - 1.1.0
           - 1.0.0
           - 0.36.1
         include:
-          - experimental: true
-            os: ubuntu-latest
+          - stable: false
             crystal: nightly
-    runs-on: ${{ matrix.os }}
-    continue-on-error: ${{ matrix.experimental }}
+    name: "crystal: ${{ matrix.crystal }}, stable: ${{ matrix.stable }}"
+    runs-on: ubuntu-latest
     container: crystallang/crystal:${{ matrix.crystal }}-alpine
+    continue-on-error: ${{ !matrix.stable }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run tests
+        run: crystal spec -v --error-trace
+
+  publish:
```

Comment:
Could this happen as a different workflow based on a tag trigger?

Reply:
Is there any benefit? I don't see one

Reply:
Completely subjective, but it provides a little cleaner separation of CI tasks.

It makes sense to run `test` via the push and schedule triggers, this could then run via [`create`](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#create) or [`release`](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release).

Reply:
I’m happy as is